src/services/broadcastdiscoveryservice.h

```diff
@@ -76,8 +77,18 @@ class BroadcastDiscoveryService :
 
 	RsPeers& mRsPeers;
 
-	RsBroadcastDiscoveryResult createResult(
-	        const UDC::IpPort& ipp, const std::string& uData );
+	/**
+	 * @brief Create result object from data
+	 * @param[in] ipp peer IP and port
+	 * @param[in] uData serialized data associated to the peer
+	 * @param[out] ec Optional storage for eventual error code,
+	 *	meaningful only on failure, if a nullptr is passed ther error is treated
+	 *	as fatal downstream, otherwise it bubble up to be treated upstream
+	 * @return nullptr on failure, pointer to the generated result otherwise
+	 */
+	static std::unique_ptr<RsBroadcastDiscoveryResult> createResult(
+	        const UDC::IpPort& ipp, const std::string& uData,
+	        rs_view_ptr<std::error_condition> ec = nullptr );
```

Comment:
That looks like a complicated construct for something that could simply be std::error_condition*

Reply:
it is a decorated way of writing `std::error_condition*` the decoration is just to informa the pointer lifetime is handled by the caller but it is just a pointer

Reply:
ah yes. I always forget that ;-)